mirrormanager2/views.py

```diff
@@ -70,6 +71,17 @@ def list_mirrors(p_name=None, p_version=None, p_arch=None):
         if arch:
             arch_id = arch.id
 
+    default_page_size = flask.current_app.config["ITEMS_PER_PAGE"]
+    try:
+        page_number = int(flask.request.args.get("page_number"))
+    except (TypeError, ValueError):
+        page_number = 1
+
+    try:
+        page_size = int(flask.request.args.get("page_size", default_page_size))
+    except (TypeError, ValueError):
+        page_size = default_page_size
+
```

Comment:
If you ever want to reuse the `PaginationArgs` dataclass in another view, you'll have to repeat this conversion block. So I would put that in the constructor of the `PaginationArgs` class (I think you can do something like that with dataclasses). The aim would be to just pass it `flask.request.args.get("page_number")` and `flask.request.args.get("page_size")` and let it do the conversion.
Since it's already a flask-specific class, we could even go as far as letting it set those attributes itself, by having it inspect `flask.request`. This way we'll have less repetition of code without sacrificing the abstraction layer! :-)

Reply:
Thanks yeah i thought about this but didnt investigate if it was possible with dataclasses -- looked at using __post_init__ in dataclasses, but since we are really just setting ti all ourselves in the dataclass itself, i just defined my own __init__ in the dataclass with the logic that previously was in the veiw endpoint.

also added a {posargs} to the lint env in tox so i can `-- --fix`